### PR TITLE
Client random

### PR DESCRIPTION
**Pull request by hardening**
We need the client_random to enable client reconnection when the client uses RDP security.

### CODE REVIEW

**Comment by hardening** [comment]
@freerdp-bot come on if you stop to test after 4s on windows for sure it will fail. Do it again !

**Comment by mfleisz** [comment]
@freerdp-bot test

**Comment by mfleisz** [comment]
@freerdp-bot test

File: libfreerdp/core/connection.c [hunk 1]
```diff
@@ -583,7 +583,14 @@ BOOL rdp_server_establish_keys(rdpRdp* rdp, wStream* s)
...
 	mod = rdp->settings->RdpServerRsaKey->Modulus;
 	priv_exp = rdp->settings->RdpServerRsaKey->PrivateExponent;
-	crypto_rsa_private_decrypt(crypt_client_random, rand_len - 8, key_len, mod, priv_exp, client_random);
+	if (crypto_rsa_private_decrypt(crypt_client_random, rand_len - 8, key_len, mod, priv_exp, client_random) <= 0)
+	{
+		free(client_random);
+		goto end;
+	}
+
+	rdp->settings->ClientRandom = client_random;
```
Comment (hardening) [inline libfreerdp/core/connection.c]: Very good question, checking that.

Comment (hardening) [inline libfreerdp/core/connection.c]: And the answer is yes, as server-side it's the only place where it is set. And this function is never called again even during a re-negotiation.

**Comment by hardening** [comment]
Feel free to merge this one if it is ok ;)

**Pull request merged**